connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/GrpcExceptionConverter.scala

```diff
@@ -230,55 +230,75 @@ private[client] object GrpcExceptionConverter {
         params.cause)),
     errorConstructor(params =>
       new NoSuchTableException(params.errorClass.orNull, params.messageParameters, params.cause)),
-    errorConstructor[NumberFormatException](params =>
+    errorConstructor(params =>
       new SparkNumberFormatException(
-        params.message,
-        params.errorClass,
+        params.errorClass.orNull,
         params.messageParameters,
         params.queryContext)),
+    errorConstructor[NumberFormatException](params =>
```

Comment:
Before this PR, we converted Java exception to Spark exception at the client side by passing the message as is (error class and message parameters was null or empty) like:  `NumberFormatException("invalid format 9hh99")` -> `SparkNumberFormatException("invalid format 9hh99", errorClass = null)`. 

After the changes, the conversion still exists but I put the original message to a parameter:
`NumberFormatException("invalid format 9hh99")` -> `SparkNumberFormatException(errorClass = "_LEGACY_ERROR_TEMP_3104", messageParameters = Map("message" -> "invalid format 9hh99"))`

Reply:
Eventually, we should convert all Java exception to Spark exception.